### DISCUSSION TITLE
resilience4j-ratelimiter not compatible with Java 11?

### DISCUSSION THREAD

**Discussion by HarishA97** [resilience4j/resilience4j#1937] [Q&A]
Hi, we had been using resilience4j-ratelimiter v1.5.0 on Java 8 for around 3 years now. We switched to Java 11 now and are seeing that it is incompatible with Java 11. We get the following error even though we have not changed anything w.r.t resilience4j-ratelimiter. Is it not possible to use resilience4j-ratelimiter with Java 11?

Our codebase is in scala version 2.11.12 and sbt 1.3.9

java.lang.IncompatibleClassChangeError: Method 'io.github.resilience4j.ratelimiter.RateLimiter io.github.resilience4j.ratelimiter.RateLimiter.of(java.lang.String, io.github.resilience4j.ratelimiter.RateLimiterConfig)' must be InterfaceMethodref constant

For people using Java 11, which version is the recommended?

**Comment by HarishA97** [comment]
I also tried upgrading resilience4j.ratelimiter to 2.x and that throws a different error

java.lang.UnsupportedClassVersionError: io/github/resilience4j/ratelimiter/RateLimiterConfig has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0

which is understandable given that the official support of 2.x is for Java 17 and I am on Java 11.

**Reply by RobWin** [reply]
Hi, usually Resilience4j 1.7 which was compiled with Java 8 should also work with Java 11.
